frontend/src/containers/ImageContainer.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { PureComponent } from 'react';
+
+import { Col, Row } from 'antd';
+
+import ImageComponent from '../components/ImageComponent';
+import { getImages } from '../data/ImagesClient';
+import Image from '../models/Image';
+
+interface Props {}
+
+interface State {
+  images: Array<Image>;
+}
+
+class ImageContainer extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = { images: [] };
+  }
+
+  componentDidMount() {
+    getImages().then((result) => {
+      this.setState({
+        images: result,
+      });
+    });
+  }
+
+  render() {
+    return (
+      <>
+        <Row gutter={[24, 24]}>
+          {this.state.images.map((image) => (
+            <Col key={image.href} span={8}>
+              <ImageComponent image={image} />
+            </Col>
+          ))}
+        </Row>
+      </>
+    );
```

Comment:
Is "clean" as far as React is concerned? I'm not really sure how to balance for-loop complexity stuff in `render()`. Should it be pulled out or is it fine to keep it as is if it's _simple-ish_

Reply:
Yeah, that looks good to me (though the fragment wrapping the `Row` isn't necessary anymore). My personal opinion for the for-loops in the render function is that it's fine as long as it's just a `map` - anything that's more complicated than a simple mapping probably deserves its own function.